[ENH] Add "chunk" entity to MRI datatype

valosekj opened the pull request:
The `chunk` entity was originally added for the 'Microscopy' datatype in BEP031 https://github.com/bids-standard/bids-specification/pull/881. This PR adds the `chunk` entity also to the MRI datatype. In the context of MRI, `chunks` refer to individual field-of-views (FOVs), for example, one FOV for the brain and the second FOV for the spinal cord.
Since two different FOVs might be acquired not only for anatomical MRI, I added the `chunk` entity also for DWI and fMRI.

Resolves: #1382

Remi-Gau commented:
@valosekj would it be possible to work on an example dataset to add to bids-example like it was done when the task entity was added for anat: https://github.com/bids-standard/bids-examples/issues/341?

effigies commented on src/schema/objects/metadata.yaml:
```diff
@@ -404,6 +404,8 @@ ChunkTransformationMatrix:
     scaling along the second and third axis respectively).
     Note that non-spatial dimensions like time and channel are not included in the
     transformation matrix.
+    In case of MRI data, the ChunkTransformationMatrix corresponds to the NIFTI [qform matrix]
+    (https://nifti.nimh.nih.gov/nifti-1/documentation/nifti1fields/nifti1fields_pages/qsform.html)
```
Currently, this is only going to show up in the microscopy section, which does not seem very helfpul. If we want to this to be applicable to MRI data, then we need to add it to their sidecars.

I'm also unclear on what this means. Does this mean that ChunkTransformationMatrix must be equal to the the qform? Or that the qform is used and therefore `ChunkTransformationMatrix` does not apply to NIfTI images? (If so, then we should just remove this addition.)

Does this need a rule, such that `qform_code != 0` (or `== 1`)? What about `sform`? Can that be anything, or must that be identical to the `qform`, or have `sform_code == 0`?

jcohenadad replied:
Agreed with @effigies, there is no need to duplicate the already existing information from the qform. So in the case of MRI, I would suggest to have this value set to `n/a` maybe?

effigies replied:
ChunkTransformationMatrix already isn't defined for MRI, so no need to even call it `n/a`. It can just be absent.

valosekj replied:
Okay! I removed the MRI paragraph from the ChunkTransformationMatrix description in https://github.com/bids-standard/bids-specification/commit/d1e4f7c8a2b5d91c56805384a7e9e5f805257438.

effigies commented on src/schema/objects/entities.yaml:
```diff
@@ -51,9 +51,12 @@ chunk:
-    The `chunk-<index>` key/value pair is used to distinguish between different regions,
-    2D images or 3D volumes files,
-    of the same physical sample with different fields of view acquired in the same imaging experiment.
+    The `chunk-<index>` entity is used to distinguish between different regions, 2D images (for example Microscopy),
+    3D volumes files, or 4D volumes files (for example DWI) of the same physical sample with different fields of
+    view acquired in the same imaging experiment.
+
+    In case of MRI data, this entity is used to distinguish between different anatomical regions, for example brain
+    and spinal cord.
```
What do you think of this wording?

```suggestion
    The `chunk-<index>` key/value pair is used to distinguish between images of
    the same physical sample with different fields of view acquired in the same
    imaging experiment.
    This entity applies to collections of 2D images, 3D volumes or 4D volume series
    (for example, diffusion weighted images), and may be used to indicate different
    anatomical structures or regions of the same structure.
```

effigies reviewed (commented):
Broadly makes sense. Do we want to bring over any of the sample metadata (such as `BodyPart`) from micr? If the main purpose in MRI is to distinguish spinal cord from brain, annotating that consistently would presumably be helpful.

https://github.com/bids-standard/bids-specification/blob/e1f853873036e2079d39a4fcb1357a04c8afbb80/src/schema/rules/sidecars/micr.yaml#L46-L69

It would go into https://github.com/bids-standard/bids-specification/blob/master/src/schema/rules/sidecars/mri.yaml.

Also, would this apply to fieldmap files as well?

Remi-Gau commented:
Note. Bringing some of the metadata like bodypart may be easier to do on its own and done via a separate PR.

effigies commented:
See #1593.

valosekj commented:
> @valosekj would it be possible to work on an example dataset to add to bids-example like it was done when the task entity was added for anat: [bids-standard/bids-examples#341](https://github.com/bids-standard/bids-examples/issues/341)?

Sure! Should I add the example dataset to this PR or create a separate PR?

effigies commented:
Examples should be submitted as PRs to https://github.com/bids-standard/bids-examples/.

Please see https://github.com/bids-standard/bids-examples/blob/master/CONTRIBUTING.md for guidance.

valosekj commented:
> @valosekj would it be possible to work on an example dataset to add to bids-example like it was done when the task entity was added for anat: [bids-standard/bids-examples#341](https://github.com/bids-standard/bids-examples/issues/341)?

Example dataset added in https://github.com/bids-standard/bids-examples/pull/405

valosekj commented:
@effigies and @Remi-Gau, is anything else required, or can this PR be merged? 
Note that I will bring some of the metadata like bodypart in a separate PR, as proposed in https://github.com/bids-standard/bids-specification/pull/1586#issuecomment-1688316196.

Remi-Gau commented:
I think that @effigies [comment](https://github.com/bids-standard/bids-specification/pull/1586#pullrequestreview-1589549811) is worth considering:

> would this apply to fieldmap files as well?

Don't know how much sense it would make for ASL but for fieldmap it may make sense to support chunck: even if in theory we could just rely on the `IntendedFor` field to say that a given fieldmap is for this or that chunk it may make it more explicit if we allow this in the filename.

jcohenadad commented:
> Don't know how much sense it would make for ASL but for fieldmap it may make sense to support chunck: even if in theory we could just rely on the IntendedFor field to say that a given fieldmap is for this or that chunk it may make it more explicit if we allow this in the filename.

Yes, good point. It is possible to acquire different chunks of EPIs, with their associated B0 fieldmap.

valosekj commented:
> I think that @effigies [comment](https://github.com/bids-standard/bids-specification/pull/1586#pullrequestreview-1589549811) is worth considering:
> 
> > would this apply to fieldmap files as well?
> 
> Don't know how much sense it would make for ASL but for fieldmap it may make sense to support chunck: even if in theory we could just rely on the `IntendedFor` field to say that a given fieldmap is for this or that chunk it may make it more explicit if we allow this in the filename.

Good point! I added the `chunk` entity to fieldmaps in https://github.com/bids-standard/bids-specification/pull/1586/commits/901fe7fa621b992e17e07c397562cfe4fe6bd79d.
Also, I added `chunk` to multiecho in https://github.com/bids-standard/bids-specification/pull/1586/commits/f6282cc9dee0026d4055ba3bd68da22f5f1e1c19 (I originally missed it).

Remi-Gau commented:
> Should this be extended to ASL images as well?

I am struggling to imagine how ASL on several chunks would be done in a way that "make sense", but one could also say the same for func. Also I have been known to lack imagination... 
So I am fine either way.

> We should not hold up on that.

Whether it should be added or not should not stop from merging this PR now.

effigies commented:
For BOLD and ASL, I think you would stitch together statistical results, not individual volumes.
